status-handler.js

```diff
@@ -143,6 +151,96 @@ class StatusHandler {
     this.log.info(`Ignored status event, no build occurred`, data);
   }
 
+  /**
+   * Check if there are third-party apps that
+   * subscribed for webhooks for a specific package
+   *
+   * @function _shouldSendWebhook
+   * @param {String} pkg - The name of the package
+   * @returns {Boolean} the computed value
+   */
+  _shouldSendWebhook(pkg) {
+    const webhooks = this.webhooks[pkg];
+    if (!webhooks || webhooks.length === 0) {
+      this.log.info(`No webhooks for pkg ${pkg}`);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Check if a build is in a queued status
+   *
+   * @function _isBuildQueued
+   * @param {Object} build - Build params
+   * @param {String} build.pkg - The package name
+   * @param {String} build.version - The package version
+   * @param {String} build.env - The build environnement
+   * @returns {String} the package name
+   */
+  async _isBuildQueued(build) {
+    // Fetch list of previous events for a certain build
+    const previousEvents = await this.models.Status.findAll(build);
+
+    // Ensure that last event is the 'queued' event from carpentd
+    const lastEvent = previousEvents[previousEvents.length - 1];
+    if (lastEvent && lastEvent.message === 'Builds Queued') return true;
+    return false;
+  }
+
+  /**
+   * Get package name from NSQ message
+   *
+   * @function _getPackageName
+   * @param {Object} data - Message from NSQ
+   * @returns {String} the package name
+   */
+  _getPackageName(data) {
+    return data.pkg || data.name;
+  }
+
+  /**
+   * Process and dispatch a webhook
+   *
+   * @function _dispatchWebhook
+   * @param {String} event - Webhook event type
+   * @param {Object} data - Message from NSQ
+   * @returns {Promise} to resolve
+   */
+  async _dispatchWebhook(event, data) {
```

Comment:
Do we need to deal with sanitizing the data, are there any secrets in the payloads here?

Reply:
No that I am aware of. @jcrugzz ?

Reply:
Not that I am aware of. @jcrugzz can you confirm?

Reply:
No secret data

Reply:
👍